### ISSUE TITLE
How to override de logic on the Basic create, retrieve, update and delete (CRUD) types and mutations

### ISSUE THREAD

**Issue by sisocobacho** [strawberry-graphql/strawberry-graphql-django#165]
I will like to override the logic inside basic create much like in django restframework modelviewsets or serializers for example. What is the best way of doing this?

**Comment by bellini666** [comment]
Hi @sisocobacho ,

What exactly are you trying to override?

You can probably override the field to call your own function instead of the ones the current ones call for example. Or you can have a basic mutation and call the create/retrieve/update/delete functions for example.

If you want to give an example of a use case I can suggest something for you

**Comment by sisocobacho** [comment]
> 

Hello, Thanks for the suggestions. Basically what I am trying to do is insert business logic inside those mutation without having to rewrite a new one from 0 and make use. off the functionality that those mutations already have. For example doing something before or after saving the model in the create.

**Comment by nrbnlulu** [comment]
Yeah hooks are a great idea..

**Comment by arseniid** [comment]
Hello everyone, are there any updates on the issue?

As an example use-case I would be interested in having something similar to:
```python
@strawberry.type
class Mutation:
    @strawberry_django.mutation(description="Create my model.")
    def create_my_model(self, data: MyModelInput, info) -> MyModel:
        create_mutation = strawberry_django.mutations.create(MyModelInput)
        my_model = create_mutation(...)
        # additional logic goes here...
```

Since this issue only has 'documentation' label, I suppose this is already possible? If yes, would you please provide a simple code snippet on how to achieve it? Thank you in advance!

**Comment by bellini666** [comment]
@arseniid yes, that should be possible. It is unfortunatly missing documentation, but in the meantime you can take a look at how [DjangoCreateMutation](https://github.com/strawberry-graphql/strawberry-django/blob/main/strawberry_django/mutations/fields.py#L227) works. Nut a quick example would be:

```python
from strawberry_django.mutations import resolvers


@strawberry.type
class Mutation:
    @strawberry_django.mutation(description="Create my model.")
    def create_my_model(self, data: MyModelInputType, info: Info) -> MyModelType:
        input_data = resolvers.parse_input(info, vars(data))
        my_django_model = resolvers.create(
            info,
            MyModel,
            input_data,
        )
        # additional business logic goes here...
```

`resolvers.parse_input` and `resolvers.create` have some extra inputs that you might be interested. Also, the original code disables the `DjangoOptimizerExtension` while creating the model, which is optional and it is not a big deal if you don't do that.

The same goes for `resolvers.update` and `resolvers.delete`

**Comment by arseniid** [comment]
@bellini666 thanks a lot for the quick response - works like a charm! I was missing the `resolvers` entry point here.
